src/NUnitFramework/tests/Api/FrameworkControllerTests.cs

```diff
@@ -132,63 +132,89 @@ public void InsertSettingsElement_SettingIsValue_CreatesASettingElementPerKey()
         {
             var outerNode = new TNode("test");
             var testSettings = new Dictionary<string, object>
             {
                 ["key1"] = "value1",
-                ["key2"] = "value2"
+                ["key2"] = value
             };
 
             var inserted = FrameworkController.InsertSettingsElement(outerNode, testSettings);
 
-            Assert.That("key1", Is.EqualTo(inserted.ChildNodes[0].Attributes["name"]));
-            Assert.That("value1", Is.EqualTo(inserted.ChildNodes[0].Attributes["value"]));
+            Assert.That(inserted.ChildNodes[0].Attributes["name"], Is.EqualTo("key1"));
+            Assert.That(inserted.ChildNodes[0].Attributes["value"], Is.EqualTo("value1"));
 
-            Assert.That("key2", Is.EqualTo(inserted.ChildNodes[1].Attributes["name"]));
-            Assert.That("value2", Is.EqualTo(inserted.ChildNodes[1].Attributes["value"]));
+            Assert.That(inserted.ChildNodes[1].Attributes["name"], Is.EqualTo("key2"));
+            Assert.That(inserted.ChildNodes[1].Attributes["value"], Is.EqualTo(value));
         }
 
-        [Test]
-        public void InsertSettingsElement_SettingIsDictionary_CreatesEntriesForDictionaryElements()
+        [TestCaseSource(nameof(SettingsData))]
+        public void InsertSettingsElement_SettingIsDictionary_CreatesEntriesForDictionaryElements(string value)
         {
             var outerNode = new TNode("test");
             var testSettings = new Dictionary<string, object>
             {
-                ["outerkey"] = new Dictionary<string, object> { { "key1", "value1" }, { "key2", "value2" } } 
+                ["outerkey"] = new Dictionary<string, object> { { "key1", "value1" }, { "key2", value } }
             };
 
             var inserted = FrameworkController.InsertSettingsElement(outerNode, testSettings);
             var settingNode = inserted.FirstChild;
-            
-            Assert.That(2, Is.EqualTo(settingNode.ChildNodes.Count));
+
+            Assert.That(settingNode.ChildNodes.Count, Is.EqualTo(2));
         }
 
-        [Test]
-        public void InsertSettingsElement_SettingIsDictionary_CreatesEntriesWithKeysAndValuesFromDictionary()
+        [TestCaseSource(nameof(SettingsData))]
+        public void InsertSettingsElement_SettingIsDictionary_CreatesValueAttributeForDictionaryElements(string value)
+        {
+            var outerNode = new TNode("test");
+            var testSettings = new Dictionary<string, object>
+            {
+                ["outerkey"] = new Dictionary<string, object> { { "key1", "value1" }, { "key2", value } }
+            };
+
+            var inserted = FrameworkController.InsertSettingsElement(outerNode, testSettings);
+            var settingNode = inserted.FirstChild;
+
+            Assert.That(settingNode.ChildNodes.Count, Is.EqualTo(2));
+            Assert.That(settingNode.Attributes["value"], Is.EqualTo($"[key1, value1], [key2, {value}]"));
```

Comment:
This is the main test for the change.

Reply:
Do we need the `ChildNodes.Count` test? It seems like that assert matches the test name `CreatesEntriesWithKeysAndValuesFromDictionary` better.

Reply:
Oh wait, it's also in `CreatesEntriesForDictionaryElements` already.

Reply:
@rprouse No biggie =)